### PR TITLE
Add debug mode to the upgrade script

### PR DESCRIPTION
**Pull request by amitaekbote**
## High-level description

This PR adds a new flag to the upgrade script `--verbose`. 

```
amita@amita-desktop:~/Testing/verbosity/genconf/serve/upgrade/d65e4416634b4acabeefc8d1f6731fb0$ bash dcos_node_upgrade.sh --verbose
Verbose mode on
This script must be run as root

amita@amita-desktop:~/Testing/verbosity/genconf/serve/upgrade/d65e4416634b4acabeefc8d1f6731fb0$ bash dcos_node_upgrade.sh --skip-checks
Skipping checks
This script must be run as root

amita@amita-desktop:~/Testing/verbosity/genconf/serve/upgrade/d65e4416634b4acabeefc8d1f6731fb0$ bash dcos_node_upgrade.sh 
This script must be run as root

amita@amita-desktop:~/Testing/verbosity/genconf/serve/upgrade/d65e4416634b4acabeefc8d1f6731fb0$ bash dcos_node_upgrade.sh --skip-checks --verbose
Skipping checks
Verbose mode on
This script must be run as root
```

## Corresponding DC/OS tickets (obligatory)

[DCOS_OSS-1658](https://jira.mesosphere.com/browse/DCOS_OSS-1658)

## Testing

Tested by adding verbose flag in the our test suite - https://github.com/mesosphere/advanced-tests/pull/48 

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label  Work In Progress

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
TC run with verbosity - https://teamcity.mesosphere.io/viewQueued.html?itemId=903228

File: dcos_installer/upgrade.py [hunk 1]
```diff
@@ -50,12 +50,20 @@
...
 SKIP_CHECKS=false
+VERBOSE=false
 
 if [[ $# -ne 0 ]]; then
-  if [[ "$1" = "--skip-checks" ]]; then
-     echo "Skipping checks"
-     SKIP_CHECKS=true
-  fi
+    for var in "$@"
+    do
```
Comment (branden) [inline dcos_installer/upgrade.py]: Let's make this `for var in "$@"; do` to be consistent with the style in the rest of this script.

File: dcos_installer/upgrade.py [hunk 2]
```diff
@@ -50,12 +50,20 @@
...
+        if [[ "$var" = "--skip-checks" ]]; then
+          echo "Skipping checks"
+          SKIP_CHECKS=true
+        fi
+        if [[ "$var" = "--verbose" ]]; then
+          echo "Debugging mode on"
```
Comment (branden) [inline dcos_installer/upgrade.py]: Let's print `"Verbose mode on"` to be consistent with the name of the flag.

File: dcos_installer/upgrade.py [hunk 3]
```diff
@@ -111,8 +119,13 @@
...
 echo "Upgrading DC/OS $role_name {{ installed_cluster_version }} -> {{ installer_version }}"
-pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }} > /dev/null
-pkgpanda activate --no-block {{ cluster_packages }} > /dev/null
+if [[ "$VERBOSE" = "true" ]]; then
+    pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }}
+    pkgpanda activate --no-block {{ cluster_packages }}
+else
+    pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }} > /dev/null
+    pkgpanda activate --no-block {{ cluster_packages }} > /dev/null
+fi
```
Comment (branden) [inline dcos_installer/upgrade.py]: Duplicating the Pkgpanda commands here feels a little hacky, and it's going to blow up the length of this script -- that `cluster_packages` var is pretty long. We could do this conditional redirect by sending output to a new file descriptor, and conditionally attaching that file descriptor to either stdout or `/dev/null`. Something like this:

```
<snip>

if [[ "$VERBOSE" = "true" ]]; then
    exec 3>&1
else
    exec 3>/dev/null
fi

<snip>

pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }} >&3
pkgpanda activate --no-block {{ cluster_packages }} >&3

<snip>
```

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

File: dcos_installer/upgrade.py [hunk 4]
```diff
@@ -50,12 +50,20 @@
...
 SKIP_CHECKS=false
+VERBOSE=false
 
 if [[ $# -ne 0 ]]; then
-  if [[ "$1" = "--skip-checks" ]]; then
-     echo "Skipping checks"
-     SKIP_CHECKS=true
-  fi
+    for var in "$@";
+    do
```
Comment (branden) [inline dcos_installer/upgrade.py]: This `do` should be on the preceding line in order to be consistent with the rest of the script.

File: dcos_installer/upgrade.py [hunk 5]
```diff
@@ -50,12 +50,20 @@
...
+        if [[ "$var" = "--skip-checks" ]]; then
+          echo "Skipping checks"
+          SKIP_CHECKS=true
+        fi
+        if [[ "$var" = "--verbose" ]]; then
+          echo "Verbose mode on"
+          VERBOSE=true
+        fi
+    done
```
Comment (branden) [inline dcos_installer/upgrade.py]: The indentation in this stanza is inconsistent, it's a mix of 4- and 2-space indents. Should be 4-space indents to be consistent with the rest of the script.

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@branden updated

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label Ready for Review

**Comment by amitaekbote** [comment]
From https://teamcity.mesosphere.io/viewLog.html?buildId=905337&buildTypeId=DcOs_Open_ManualTriggers_UpgradeTest_FromStable&tab=buildLog&_focus=145#_state=145 

```
[20:03:40][Step 5/6] [2017-12-19 20:03:40,135|dcos_test_utils.ssh_client|DEBUG]: Running socket cmd: /usr/bin/ssh -oConnectTimeout=10 -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oLogLevel=ERROR -oBatchMode=yes -oPasswordAuthentication=no -oControlPath=/teamcity/temp/buildTmp/tmptbdx3cf5 -oControlMaster=auto -p 22 centos@34.216.214.179 sudo bash dcos_node_upgrade.sh --verbose
[20:03:40][Step 5/6] Verbose mode on
[20:03:40][Step 5/6] Upgrading DC/OS master 1.10.2 -> 1.11-dev
[20:03:41][Step 5/6] Fetching: adminrouter--e65a65c30b0dbeb65d58abbb5b02dcff1e3e9761
[20:03:41][Step 5/6] Fetched: adminrouter--e65a65c30b0dbeb65d58abbb5b02dcff1e3e9761
[20:03:41][Step 5/6] Fetching: avro-cpp--b94c4ed2788450b73d454794585fbf169e3b844f
[20:03:41][Step 5/6] Fetched: avro-cpp--b94c4ed2788450b73d454794585fbf169e3b844f
[20:03:41][Step 5/6] Fetching: boost-libs--a04883bd55bb23fe577a8e60b01bfb0948c18c73Package already added.
[20:03:41][Step 5/6] Fetched: boost-libs--a04883bd55bb23fe577a8e60b01bfb0948c18c73
[20:03:41][Step 5/6] Fetching: bootstrap--0d1cb352c020dc6f23d47a826e27999d571c1951
[20:03:41][Step 5/6] Fetched: bootstrap--0d1cb352c020dc6f23d47a826e27999d571c1951
[20:03:41][Step 5/6] Fetching: boto--bc7941697846321577f214aa3b973e56d19beb9d
[20:03:41][Step 5/6] Fetched: boto--bc7941697846321577f214aa3b973e56d19beb9d
[20:03:41][Step 5/6] Fetching: check-time--f1609ae2fff1ddedc1d32635dc40e6ab7d2c48b6Package already added.
[20:03:41][Step 5/6] Fetched: check-time--f1609ae2fff1ddedc1d32635dc40e6ab7d2c48b6
[20:03:42][Step 5/6] Fetching: cni--c3b4dbf04ffcb3c59ff69b62aeae6b1ddb2d2bae
[20:03:42][Step 5/6] Fetched: cni--c3b4dbf04ffcb3c59ff69b62aeae6b1ddb2d2bae
[20:03:43][Step 5/6] Fetching: cosmos--7fd57c8134ec2e6f6829056b09181bc0b4e3ac2a
[20:03:43][Step 5/6] Fetched: cosmos--7fd57c8134ec2e6f6829056b09181bc0b4e3ac2a
[20:03:43][Step 5/6] Fetching: curl--9e23dc8771748b39a8333192be70b97c924f5938
[20:03:43][Step 5/6] Fetched: curl--9e23dc8771748b39a8333192be70b97c924f5938
[20:03:43][Step 5/6] Fetching: dcos-checks--8e85c6860115e74c3f75551ee96d2e85e0b5cf25
[20:03:43][Step 5/6] Fetched: dcos-checks--8e85c6860115e74c3f75551ee96d2e85e0b5cf25
[20:03:43][Step 5/6] Fetching: dcos-cni--cc00e84332454d6c3b3d94f4752a673ba0faac82Package already added.
[20:03:43][Step 5/6] Fetched: dcos-cni--cc00e84332454d6c3b3d94f4752a673ba0faac82
[20:03:44][Step 5/6] Fetching: dcos-config--setup_adb464a40b55ac8dcbb44523827d144f7cce6f5f
[20:03:44][Step 5/6] Fetched: dcos-config--setup_adb464a40b55ac8dcbb44523827d144f7cce6f5f
[20:03:44][Step 5/6] Fetching: dcos-diagnostics--6a25720ece5b6630afd56e138da28ba0cc4dd761
[20:03:44][Step 5/6] Fetched: dcos-diagnostics--6a25720ece5b6630afd56e138da28ba0cc4dd761
[20:03:44][Step 5/6] Fetching: dcos-history--af9b4db81f5c2b445df79f871279a0cfa95b1393
[20:03:44][Step 5/6] Fetched: dcos-history--af9b4db81f5c2b445df79f871279a0cfa95b1393
```

**Comment by amitaekbote** [comment]
Without verbosity flag
https://teamcity.mesosphere.io/viewLog.html?buildId=905259&buildTypeId=DcOs_Open_ManualTriggers_UpgradeTest_FromStable&tab=buildLog&_focus=503#_state=503

```
[19:27:12][Step 5/6] [2017-12-19 19:27:12,544|dcos_test_utils.ssh_client|DEBUG]: Running socket cmd: /usr/bin/ssh -oConnectTimeout=10 -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oLogLevel=ERROR -oBatchMode=yes -oPasswordAuthentication=no -oControlPath=/teamcity/temp/buildTmp/tmp47e3k2wu -oControlMaster=auto -p 22 centos@35.165.216.172 sudo bash dcos_node_upgrade.sh
[19:27:12][Step 5/6] Upgrading DC/OS master 1.10.2 -> 1.11-dev
[19:30:06][Step 5/6] [2017-12-19 19:30:06,252|dcos_test_utils.ssh_client|DEBUG]: Closing SSH Tunnel: /usr/bin/ssh -oConnectTimeout=10 -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oLogLevel=ERROR -oBatchMode=yes -oPasswordAuthentication=no -oControlPath=/teamcity/temp/buildTmp/tmp47e3k2wu -oControlMaster=auto -p 22 -O exit centos@35.165.216.172
[19:30:06][Step 5/6] [2017-12-19 19:30:06,258|upgrade|INFO]: Upgrading master: '35.160.198.53'
[19:30:06][Step 5/6] [2017-12-19 19:30:06,258|dcos_test_utils.ssh_client|DEBUG]: Starting SSH tunnel: /usr/bin/ssh -oConnectTimeout=10 -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oLogLevel=ERROR -oBatchMode=yes -oPasswordAuthentication=no -oControlPath=/teamcity/temp/buildTmp/tmp00rral7g -oControlMaster=auto -p 22 -fnN -i /teamcity/temp/buildTmp/tmpmy3m0lon centos@35.160.198.53
[19:30:06][Step 5/6] [2017-12-19 19:30:06,350|dcos_test_utils.ssh_client|DEBUG]: SSH Tunnel established!
```

**Pull request merged**